Library/Homebrew/formula_creator.rb

```diff
@@ -5,49 +5,58 @@
 require "erb"
 
 module Homebrew
+  module_function
+
+  def url2name(url)
+    odebug "url: #{url}"
+    stem = Pathname.new(url).stem
+    odebug "stem: #{stem}"
+    # special cases first
+    name = if stem.start_with? "index.cgi"
+      # http://www.codesrc.com/gitweb/index.cgi?p=libzipper.git;a=summary
+      # https://github.com/Homebrew/brew/pull/9459
+      stem.rpartition("=").last
+    elsif url =~ %r{github\.com/\S+/(\S+)/(archive|releases)/}
+      Regexp.last_match(1)
+    else
```

Comment:
Should the `else` not be the original behaviour?

Reply:
Not sure I understand it right. `else` is the original behavior from here.

https://github.com/Homebrew/brew/blob/1b8207bf47730b6fd328ac54d59e7baa635e0b62/Library/Homebrew/formula_creator.rb#L33-L34